server/routes/offenceRoutes.ts

```diff
@@ -1165,6 +1185,62 @@ export default class OffenceRoutes {
     )
   }
 
+  public getSentenceLengthMismatch: RequestHandler = async (req, res): Promise<void> => {
+    const { nomsId, courtCaseReference, appearanceReference, addOrEditCourtCase, addOrEditCourtAppearance } = req.params
+    const courtAppearance = this.courtAppearanceService.getSessionCourtAppearance(req.session, nomsId)
+
+    const overallSentenceLengthComparison = await this.calculateReleaseDatesService.compareOverallSentenceLength(
+      courtAppearance,
+      req.user.username,
+    )
+
+    return res.render('pages/offence/sentence-length-mismatch', {
+      nomsId,
+      courtCaseReference,
+      courtAppearance,
+      appearanceReference,
+      addOrEditCourtCase,
+      addOrEditCourtAppearance,
+      overallSentenceLengthComparison,
+      isReviewOffences: true,
+      errors: req.flash('errors') || [],
+      backLink: `/person/${nomsId}/${addOrEditCourtCase}/${courtCaseReference}/${addOrEditCourtAppearance}/${appearanceReference}/offences/check-offence-answers`,
+    })
+  }
+
+  public submitSentenceLengthMismatch: RequestHandler = async (req, res): Promise<void> => {
+    const { nomsId, courtCaseReference, appearanceReference, addOrEditCourtCase, addOrEditCourtAppearance } = req.params
+
+    const sentenceLengthMismatch = trimForm<SentenceLengthMismatchForm>(req.body)
+    const errors = validate(
+      sentenceLengthMismatch,
+      {
+        confirmMismatch: 'required',
+      },
+      {
+        'required.confirmMismatch': `You must select whether you want to continue.`,
+      },
+    )
+
+    if (errors.length > 0) {
+      req.flash('errors', errors)
+      return res.redirect(
+        `/person/${nomsId}/${addOrEditCourtCase}/${courtCaseReference}/${addOrEditCourtAppearance}/${appearanceReference}/offences/sentence-length-mismatch`,
+      )
+    }
+
+    if (sentenceLengthMismatch.confirmMismatch === 'yes') {
+      this.courtAppearanceService.setOffenceSentenceAccepted(req.session, nomsId, true)
```

Comment:
I think you may always need to set this regardless of the answer? If a user navigated back to the check offences screen after selecting yes and then selects no I think this should be false

Reply:
So here "no" means that they've realised they've made a mistake because the lengths mismatch. So they're redirected back to the offences page to continue making changes.

If they select "yes" they're happy that the sentence lengths mismatch and then the section in the task list gets set to complete.